pkg/k8s/k8s.go

```diff
@@ -455,3 +462,197 @@ func (c *cluster) ClusterNameVersion() (string, string, error) {
+		if !hasRequiredData {
+			continue
+		}
+		dockerConfig := &docker.Config{}
+		err := dockerConfig.Read(data, isLegacy)
+		if err != nil {
+			return nil, fmt.Errorf("reading %s or %s field of %q secret: %w", corev1.DockerConfigJsonKey, corev1.DockerConfigKey, secret.Namespace+"/"+secret.Name, err)
+		}
+		for authKey, auth := range dockerConfig.Auths {
+			server, err := docker.GetServerFromDockerAuthKey(authKey)
+			if err != nil {
+				return nil, err
+			}
+			if a, ok := auths[server]; multiSecretSupport && ok {
+				user := fmt.Sprintf("%s,%s", a.Username, auth.Username)
+				pass := fmt.Sprintf("%s,%s", a.Password, auth.Password)
+				auths[server] = docker.Auth{Username: user, Password: pass}
+			} else {
+				auths[server] = auth
+			}
+		}
+	}
+	return auths, nil
+}
+
+type ContainerImages map[string]string
+
+func MapContainerNamesToDockerAuths(imageRef string, auths map[string]docker.Auth) (docker.Auth, error) {
+	wildcardServers := GetWildcardServers(auths)
+
+	var authsCred docker.Auth
+	server, err := docker.GetServerFromImageRef(imageRef)
+	if err != nil {
+		return authsCred, err
+	}
+	if auth, ok := auths[server]; ok {
+		return auth, nil
+	}
+	if len(wildcardServers) > 0 {
+		if wildcardDomain := matchSubDomain(wildcardServers, server); len(wildcardDomain) > 0 {
+			if auth, ok := auths[wildcardDomain]; ok {
+				return auth, nil
+			}
+		}
+	}
+
+	return authsCred, nil
+}
+
+func GetWildcardServers(auths map[string]docker.Auth) []string {
+	wildcardServers := make([]string, 0)
+	for server := range auths {
+		if strings.HasPrefix(server, "*.") {
+			wildcardServers = append(wildcardServers, server)
+		}
+	}
+	return wildcardServers
+}
+
+func matchSubDomain(wildcardServers []string, subDomain string) string {
+	for _, domain := range wildcardServers {
+		domainWithoutWildcard := strings.Replace(domain, "*", "", 1)
+		if strings.HasSuffix(subDomain, domainWithoutWildcard) {
+			return domain
+		}
+	}
+	return ""
+}
+
+func getWorkloadPodSpec(un unstructured.Unstructured) (*corev1.PodSpec, error) {
+	switch un.GetKind() {
+	case KindPod:
+		objectMap, ok, err := unstructured.NestedMap(un.Object, []string{"spec"}...)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, fmt.Errorf("unstructured resource do not match Pod spec")
+		}
+		return mapToPodSpec(objectMap)
+	case KindCronJob:
+		objectMap, ok, err := unstructured.NestedMap(un.Object, []string{"spec", "jobTemplate", "spec", "template", "spec"}...)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, fmt.Errorf("unstructured resource do not match Pod spec")
+		}
+		return mapToPodSpec(objectMap)
+	case KindDeployment:
+		objectMap, ok, err := unstructured.NestedMap(un.Object, []string{"spec", "template", "spec"}...)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, fmt.Errorf("unstructured resource do not match Pod spec")
+		}
+		return mapToPodSpec(objectMap)
+	default:
+		return nil, nil
+	}
+}
```

Comment:
Hi @chen-keinan I noticed that only Pod, CronJob and Deployment is included here. Is there a reason why the other workload types are not supported? I noticed that a [similar implementation in trivy operator](https://github.com/aquasecurity/trivy-operator/blob/7fe8c3912f5356892e7f93172939523423f1de48/pkg/kube/object.go#L239-L262) includes more workload types.

Reply:
@smtan-gl good catch up , do you wish to raise a PR ?

Reply:
good point , I'll add a fix for it